### ISSUE TITLE
Colorcoder styles broken

### ISSUE THREAD

**Issue by Sawtaytoes** [babel/babel-sublime#395]
I really appreciate the latest update to babel-sublime. It fixes a lot of issues I was having specifically related to JSX fragments and incorrect regex detection.

The main issue I've seen though is in compatibility with [Colorcoder](https://packagecontrol.io/packages/Colorcoder), a semantic-highlighting plugin. I don't know what changed in this latest release, but my code looks unusually bad now.

### babel-sublime v8
It used to look like this:

![image](https://user-images.githubusercontent.com/3948069/114254163-6a619180-9973-11eb-912f-5fa7e8c2f6f4.png)

### babel-sublime v10
Now it looks like this:

![image](https://user-images.githubusercontent.com/3948069/114254124-3be3b680-9973-11eb-842a-f062c8caef04.png)

**Comment by Thom1729** [comment]
I wasn't familiar with Colorcoder before, so I read the documentation. It looks like it's specifically designed to work with a fixed set of syntax definitions, including the old JavaScript Next definition. The new definition is partly descended from JavaScript Next, but with several years of improvements; it might not actually contain any of the original JavaScript Next code anymore.

I don't think this is something that can be fixed on this end; the Colorcoder plugin would have to either add special support for the new syntax or be reimplemented to work with syntaxes it wasn't explicitly designed for. Unfortunately, Colorcoder hasn't been updated since 2014 — I ran into errors testing it out because it can't handle the `sublime-color-scheme` format — and so I doubt that it will be updated.

Would something like [hashed syntax highlighting](https://www.sublimetext.com/docs/3/color_schemes.html#hashed_syntax_highlighting) work for your use case?

**Comment by Sawtaytoes** [comment]
I'm not sure what you mean about `sublime-color-scheme` format. I haven't had a theme that doesn't work with it; although, some syntax highlighting schemes don't look right.

Are you saying that's because there's something it's missing?

---

I think "hashed syntax highlighting" is what Colorcoder does under-the-hood.

You have to configure a theme to work with it by running the "Tweak Colorcoder" option:

![image](https://user-images.githubusercontent.com/3948069/114260062-28971200-9998-11eb-8e53-ac08f5dabc9c.png)

It then adds colors for every possible CRC value in the theme (yes, it's hacky):

```xml
<dict>
  <key>scope</key>
  <string>cc0xb9</string>
  <key>settings</key>
  <dict>
    <key>background</key>
    <string>#263237</string>
    <key>foreground</key>
    <string>#9167e5</string>
  </dict>
</dict>
<dict>
  <key>scope</key>
  <string>cc0xba</string>
  <key>settings</key>
  <dict>
    <key>background</key>
    <string>#263237</string>
    <key>foreground</key>
    <string>#9467e5</string>
  </dict>
</dict>
<dict>
  <key>scope</key>
  <string>cc0xbb</string>
  <key>settings</key>
  <dict>
    <key>background</key>
    <string>#263237</string>
    <key>foreground</key>
    <string>#9767e5</string>
  </dict>
</dict>
```

Then when using the theme, it's running CRC hashes against variables and letting the theme choose a color for it.

Does it need something special for it to know what to CRC hash?

---

I don't mind forking Colorcoder if I can get this working; although, I'd need a bit of help understanding what I need to look for to make it compatible with the latest v10 of babel-sublime.

I'm a JS dev and use only Sublime Text because no other editor provides semantic highlighting; otherwise, I'd most-likely be on VS Code by now. I feel very handicapped by it's exclusion from literally every other editor.

I see this as an accessibility plugin making code far easier for me to read. If you could help me figure out what I need to do, I'd greatly appreciate it.

It's the only killer-app feature I won't go without to the degree that I even wrote an article on it hoping to draw attention to the feature:
https://medium.com/@Sawtaytoes/why-i-dont-use-visual-studio-code-f5ac7274fb96

**Comment by Thom1729** [comment]
> I'm not sure what you mean about sublime-color-scheme format. I haven't had a theme that doesn't work with it; although, some syntax highlighting schemes don't look right.

> Are you saying that's because there's something it's missing?

When I tried Colorcoder, there were some parsing errors that I interpreted to mean that it didn't support the newer [.sublime-color-scheme format](https://www.sublimetext.com/docs/3/color_schemes.html). I'm fairly sure that `.sublime-color-scheme` is newer than 2014, so this is not surprising. It worked for me when I used an old `.tmTheme` color scheme. I may have missed something; as I said, I've never used Colorcoder before yesterday.

I suspect that Colorcoder may need a major update. I haven't read through its source code, so I'm not sure exactly how it works. I'm pretty sure it doesn't use Sublime's built-in hashed syntax highlighting because that's also newer than 2014. In fact, it looks like Colorcoder itself is basically a really complicated way of implementing the same feature as hashed syntax highlighting. If you're not familiar with Sublime's built-in implementation, definitely check out [the documentation](https://www.sublimetext.com/docs/3/color_schemes.html#hashed_syntax_highlighting) to see if that could simply replace Colorcoder in your workflow..
